3drepo/3drepo.io
# ISSUE #3824 - V5 Repeated calls to containers endpoint when looking at Federations

This fixes #3824

#### Description

This was caused by the `useContainersData()` in the edit federation modal. By changing the render method of this modal to use the new `DialogsActionsDispatchers.open()` method the component only renders (thus `useContainersData()` is called) when the modal is opened.
Some changes were made to how the teamspace and project are fetched since `useParams()` doesn't work with the action dispatching method of opening modals. Instead I have used redux selectors to get these values. I think this shouldn't cause issues, but if you think otherwise do say.
I also fixed a small problem where the save changes button would be disabled if there were no containers included

#### Test cases

- open up devtools with the Network tab open.
- Go to the federations list and look to see if there are any unnecessary fetchContainers calls
- Open the editFederation modal and check it all works fine
